### PR TITLE
LDAP integration tests

### PR DESCRIPTION
**Pull request by ggsdc**
Integration test for the ldap authentication

### CODE REVIEW

File: cornflow/config.py [hunk 1]
```diff
@@ -54,15 +54,24 @@ class Testing(object):
...
+    LDAP_GROUP_OBJECT_CLASS = os.getenv("LDAP_GROUP_OBJECT_CLASS")
+    LDAP_GROUP_ATTRIBUTE = os.getenv("LDAP_GROUP_ATTRIBUTE")
+    LDAP_GROUP_BASE = os.getenv("LDAP_GROUP_BASE")
+    LDAP_GROUP_TO_ROLE_SERVICE = os.getenv("LDAP_GROUP_TO_ROLE_SERVICE")
+    LDAP_GROUP_TO_ROLE_ADMIN = os.getenv("LDAP_GROUP_TO_ROLE_ADMIN")
+    LDAP_GROUP_TO_ROLE_VIEWER = os.getenv("LDAP_GROUP_TO_ROLE_VIEWER")
+    LDAP_GROUP_TO_ROLE_PLANNER = os.getenv("LDAP_GROUP_TO_ROLE_PLANNER")
+
+    LDAP_PROTOCOL_VERSION = int(os.getenv("LDAP_PROTOCOL_VERSION", 3))
+    LDAP_USE_TLS = os.getenv("LDAP_USE_TLS")
```
Comment (pchtsp) [inline cornflow/config.py:74]: if you're using environment variables, then just make `Testing` inherit from `MainEnvVars` as the other two. You do not need these lines then

File: cornflow/tests/custom_test_case.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 This file contains the different custom test classes used to generalize the unit testing of cornflow.
 """
 # Import from libraries
+from abc import ABC
```
Comment (pchtsp) [inline cornflow/tests/custom_test_case.py:5]: are you using this? I do not see any reference to it.

File: cornflow/tests/custom_test_case.py [hunk 3]
```diff
@@ -426,3 +427,189 @@ def test_payload_bad_format(self):
...
+
+            self.assertEqual(400, response.status_code)
+            self.assertEqual(
+                "Invalid token, please try again with a new token",
+                response.json["error"],
+            )
+
+        def test_token(self):
+            # TODO: implement to check correct token creation
+            pass
```
Comment (pchtsp) [inline cornflow/tests/custom_test_case.py:615]: this is missing?

File: cornflow/tests/ldap/test_ldap_authentication.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            self.response.json["id"],
+        )
+
+    def test_role_registration(self):
+        RegisterRoles().run()
+        super().test_successful_log_in()
+        user_role = UserRoleModel.query.filter_by(
+            user_id=self.response.json["id"], role_id=PLANNER_ROLE
+        )
+        self.assertNotEqual(None, user_role)
```
Comment (pchtsp) [inline cornflow/tests/ldap/test_ldap_authentication.py:61]: do you want to also test the endpoints that need to fail? editing a username, editing / deleting an assignment, etc.

File: cornflow/tests/ldap/test_ldap_authentication.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+class TestLogIn(LoginTestCases.LoginEndpoint):
+    def setUp(self):
+        super().setUp()
+        self.AUTH_TYPE = current_app.config["AUTH_TYPE"]
+        self.data = {"email": "planner", "password": "planner1234"}
+
+    def tearDown(self):
+        super().tearDown()
+
```
Comment (pchtsp) [inline cornflow/tests/ldap/test_ldap_authentication.py:23]: you can remove this

File: cornflow/tests/unit/test_log_in.py [hunk 6]
```diff
@@ -1,220 +1,35 @@
...
+        user.save()
+        db.session.commit()
+        # we take out the name, we do not need it to sign in
+        self.data.pop("name")
+        self.id = UserModel.query.filter_by(name="testname").first().id
 
     def tearDown(self):
-        db.session.remove()
-        db.drop_all()
+        super().tearDown()
```
Comment (pchtsp) [inline cornflow/tests/unit/test_log_in.py:32]: no need for this

File: cornflow/config.py (hunk 1 above)
Comment (ggsdc) [reply cornflow/config.py:74]: Will change it

File: cornflow/tests/custom_test_case.py (hunk 2 above)
Comment (ggsdc) [reply cornflow/tests/custom_test_case.py:5]: No, this was a mistake

File: cornflow/tests/custom_test_case.py (hunk 3 above)
Comment (ggsdc) [reply cornflow/tests/custom_test_case.py:628]: Is a test that is dificult to implement as the token has the timestamp on it, so I do not know how to test that I can generate a token that is the same that is generated at log in time.

**Comment by ggsdc** [comment]
Made all the changes and should be ready for review and approval.

File: cornflow/endpoints/roles.py [hunk 7]
```diff
@@ -187,7 +187,7 @@ def post(self, **kwargs):
         AUTH_TYPE = current_app.config["AUTH_TYPE"]
         if AUTH_TYPE == AUTH_LDAP:
             raise EndpointNotImplemented(
-                "The roles have to be created in the directory."
+                "The roles have to be created in the directory"
```
Comment (pchtsp) [inline cornflow/endpoints/roles.py:190]: The "role assignments", really. The "roles" are another thing.

**Review by pchtsp — APPROVED** [review]

**Pull request merged**